src/humidity.jl

```diff
@@ -0,0 +1,63 @@
+"""
+Compute the saturation specific humidity for a single atmospheric level.
+"""
+function get_saturation_specific_humidity(
+    T::Array{NF,2},  # Absolute temperature for a single atmospheric level
+    p::Array{NF,2},  # Normalised surface pressure
+    σ::NF,           # σ of the current atmospheric level
+    M::ModelSetup{NF},
+) where {NF<:AbstractFloat}
+    @unpack nlon, nlat = M.geospectral.geometry
+    @unpack e₀, C₁, C₂, T₀, T₁, T₂ = M.params.humidity_coefs
+
+    Qsat = zeros(nlon, nlat)  # Saturation specific humidity
+
+    @inbounds for j = 1:nlat, i = 1:nlon
+        if T[i, j] > T₀
+            Qsat[i, j] = e₀ * exp(C₁ * (T[i, j] - T₀) / (T[i, j] - T₁))
+        else
+            Qsat[i, j] = e₀ * exp(C₂ * (T[i, j] - T₀) / (T[i, j] - T₂))
+        end
+    end
+
+    if σ <= 0
```

Comment:
Full sigma levels are never negative, the half levels start at zero for the top of the atmosphere, see https://github.com/milankl/SpeedyWeather.jl/blob/main/input_data/vertical_coordinate/vertical_levels.png I know this is just taken from Fortran but I don't know when this branch would actually be executed?

Reply:
That's a very good point! Full sigma levels can be zero though, right?

In any case I don't really get what's going on here. And I especially wouldn't be comfortable merging this before understanding the use of `p[1, 1]` here. Do you have any idea what that could be about?

Reply:
`Qsat` gets calculated for all `k`, but is only actually used (so far at least) for `k >= 2`. So this (weird) branch may get executed once for `σ == 0`, but then the result of that calculation never gets used?

Reply:
For default resolution the full sigma levels are currently
```julia
julia> m.geospectral.geometry.σ_levels_full
8-element Vector{Float64}:
 0.036597223182346206
 0.11748566048999509
 0.2169349015956573
 0.3402683261876466
 0.4905020311609761
 0.6631496767486345
 0.8333721536349882
 0.9565026461476916
```
while the first half level is 0, and the last is 1, the full levels are the mid-points, so we'll always have 0<sigma<1.

Reply:
Ahhh I had full and half the wrong way around :)

Reply:
Still though, any idea what this actually does?

```julia
@. Qsat = 622.0 * Qsat / (σ * p - 0.378 * Qsat)
```

Reply:
Something like
![image](https://user-images.githubusercontent.com/25530332/170485338-a5a8083d-ae63-4f8f-97c9-a23a5b59e70b.png)

I believe `622.0` is the ratio of molecular weight of water (2H+O = 18g/mol) to dry air (28.96g/mol, see [here](https://socratic.org/questions/what-is-the-molecular-weight-of-air)) times 1000 for some conversion?